aiohttp/web_urldispatcher.py

```diff
@@ -1052,20 +1113,36 @@ def register_resource(self, resource: AbstractResource) -> None:
                     "already handled by {!r}".format(name, self._named_resources[name])
                 )
             self._named_resources[name] = resource
-        self._resources.append(resource)
+        if append:
+            self._resources.append(resource)
 
     def add_resource(self, path: str, *, name: Optional[str] = None) -> Resource:
         if path and not path.startswith("/"):
             raise ValueError("path should be started with / or be empty")
+        # Reuse named resource
+        if name:
+            resource = self._named_resources.get(name)
+            if resource is None:
+                pass
+            elif resource.raw_match(path):
+                return cast(Resource, resource)
+            else:
+                raise ValueError(f"Duplicate {name!r}, already handled by {resource!r}")
```

Comment:
This looks like a separate change, it doesn't seem related to the rest of the PR. It also appears to make the below section (line 1142) redundant.

Reply:
In this section reuse all resources by name.
Next section for new PlainResource or reuse without name PlainResource.
And next section for another new or reuse without name last resource.
Finally section create DynamicResource

Reply:
So, 3rd section does `if resource.name == name and resource.raw_match(path):`. But, if it has a name, wouldn't it already exist in `self._named_resources` and therefore already have done this check above? I'm struggling to see a situation where this 3rd check will ever run and return true.

Reply:
For example DynamicResource without name